src/frontend/src/views/Data/components/SamplesView/components/SampleTableModalManager/index.tsx

```diff
@@ -0,0 +1,115 @@
+import { useEffect, useState } from "react";
+import { CreateNSTreeModal } from "./components/CreateNSTreeModal";
+import { DeleteSamplesConfirmationModal } from "./components/DeleteSamplesConfirmationModal";
+import DownloadModal from "./components/DownloadModal";
+import { EditSamplesConfirmationModal } from "./components/EditSamplesConfirmationModal";
+import { SampleTableActions } from "./components/SampleTableActions";
+import { UsherTreeFlow } from "./components/UsherTreeFlow";
+
+// TODO-TR (mlila): types
+interface Props {
+  checkedSamples: Sample[];
+  clearCheckedSamples(): void;
+}
+
+const SampleTableModalManager = ({
+  checkedSamples,
+  clearCheckedSamples,
+}: Props): JSX.Element => {
+  const [checkedSampleIds, setCheckedSampleIds] = useState<string[]>([]);
+  const [failedSampleIds, setFailedSampleIds] = useState<string[]>([]);
+  const [canEditSamples, setCanEditSamples] = useState<boolean>(false);
+
+  const [isDeleteSampleModalOpen, setIsDeleteSampleModalOpen] =
+    useState<boolean>(false);
+  const [isEditSampleModalOpen, setIsEditSampleModalOpen] =
+    useState<boolean>(false);
+  const [isDownloadModalOpen, setIsDownloadModalOpen] =
+    useState<boolean>(false);
+  const [isNSCreateTreeModalOpen, setIsNSCreateTreeModalOpen] =
+    useState<boolean>(false);
+  const [shouldStartUsherFlow, setShouldStartUsherFlow] =
+    useState<boolean>(false);
+
+  // generate ID lists from sample objects
+  useEffect(() => {
+    const checkedIds = checkedSamples.map((s) => s.publicId);
+    const failedIds = checkedSamples
+      .filter((s) => s.CZBFailedGenomeRecovery)
+      .map((s) => s.publicId);
+
+    setCheckedSampleIds(checkedIds);
+    setFailedSampleIds(failedIds);
+  }, [checkedSamples]);
+
+  // determine whether selected samples can be edited
+  useEffect(() => {
+    const numberOfCheckedSamples = checkedSampleIds.length;
+    if (numberOfCheckedSamples > 0 && numberOfCheckedSamples <= 100) {
+      setCanEditSamples(true);
+    } else {
+      setCanEditSamples(false);
+    }
+  }, [checkedSampleIds]);
+
+  useEffect(() => {
+    if (shouldStartUsherFlow) setShouldStartUsherFlow(false);
```

Comment:
for context, it was originally like this because it was a flow that contained 3 modals whose state depended on each other (to determine which of the 3 were open). But now it's been downsized to two and it hasn't caused us problems so far, which is why we haven't changed it yet